mdoc/mdoc.Test/MDocUpdaterTests.cs

```diff
@@ -76,5 +77,59 @@ public void InternalEIITest()
             Assert.AreEqual("System.String mdoc.Test2.InternalEIICalss::mdoc.Test.SampleClasses.InterfaceA.Getstring(System.Int32)", oldmember.FullName);
 
         }
+
+        [Test]
+        public void UpdateToRight_MethodInterface_Test()
+        {
+            var member = GetType(typeof(mdoc.Test2.EiiImplementClass)).Methods.FirstOrDefault(t => t.FullName == "System.String mdoc.Test2.EiiImplementClass::GetNo()");
+            var nodeinfo = UpdateXml("GetNo", member);
+            Assert.IsTrue(nodeinfo.Count() == 1);
+            Assert.AreEqual("M:mdoc.Test2.Interface_B.GetNo", nodeinfo[0].InnerText);
+        }
+
+        [Test]
+        public void UpdateToRight_PropertyInterface_Test()
+        {
+            var member = GetType(typeof(mdoc.Test2.EiiImplementClass)).Properties.FirstOrDefault(t => t.FullName == "System.Int32 mdoc.Test2.EiiImplementClass::no()");
+            var nodeinfo = UpdateXml("no", member);
+            Assert.IsTrue(nodeinfo.Count() == 1);
+            Assert.AreEqual("P:mdoc.Test2.Interface_A.no", nodeinfo[0].InnerText);
+        }
+
+        [Test]
+        public void UpdateToRight_EventInterface_Test()
+        {
+            var member = GetType(typeof(mdoc.Test2.EiiImplementClass)).Events.FirstOrDefault(t => t.FullName == "System.EventHandler`1<System.EventArgs> mdoc.Test2.EiiImplementClass::ItemChanged");
+            var nodeinfo = UpdateXml("ItemChanged", member);
+            Assert.IsTrue(nodeinfo.Count() == 1);
+            Assert.AreEqual("E:mdoc.Test2.Interface_A.ItemChanged", nodeinfo[0].InnerText);
+        }
+
+        private List<XmlElement> UpdateXml(string XmlNodeName, MemberReference mi)
+        {
+            List<XmlElement> returnValue = new List<XmlElement>();
+
+            var type = GetType(typeof(mdoc.Test2.EiiImplementClass));
+            var ieeImplementList = MDocUpdater.GetTypeEiiMembers(type);
+            var typeInterfaces = GetClassInterface(type);
+
+            var doc = new XmlDocument();
+            doc.LoadXml(XmlConsts.EiiErrorImplement);
+
+            var node = doc.SelectSingleNode($"/Type/Members/Member[@MemberName='{XmlNodeName}']");
+
+            if (node != null)
+            {
+                BindingFlags flags = BindingFlags.NonPublic | BindingFlags.Static;
+                MethodInfo mInfo = typeof(MDocUpdater).GetMethod("AddImplementedMembers", flags);
```

Comment:
actually ... this test looks great, but I'd rather prefer that you just make the `AddImplementedMembers` method public, and invoke it directly, rather than using reflection :)